packages/block-library/src/site-title/edit/index.js

```diff
@@ -71,10 +81,11 @@ export default function SiteTitleEdit( {
 					href="#site-title-pseudo-link"
 					onClick={ ( event ) => event.preventDefault() }
 				>
-					{ readOnlyTitle || __( 'Site Title placeholder' ) }
+					{ decodeEntities( title ) ||
+						__( 'Site Title placeholder' ) }
 				</a>
 			) : (
-				<span>{ title || readOnlyTitle }</span>
+				<span>{ decodeEntities( title ) }</span>
```

Comment:
Not related to this PR, but I guess we should show `Site Title placeholder` as a fallback here too, no?

Reply:
Good call; we have a similar fallback Site Tagline block. I'm fixing the same issue there as well.